File: 17solve/17-150 var2.py
# Автор В.Н. Шубинкин

count = 0
min_s = 20000
with open('17-1.txt') as f:
    a = int(f.readline())
    for line in f.readlines():
        b = int(line)
        if (a % 7 == 0 and b % 17 != 0
            or b % 7 == 0 and a % 17 != 0):
            count += 1
            min_s = min(min_s, a + b)
        a = b
print(count, min_s)

File: 17solve/17-150 var2.pas
// Автор В.Н. Шубинкин
##
var n := 10000; // количество элементов подсмотрено в блокноте
assign(input, '17-1.txt');
var count := 0;
var minS := 20000;
var a := ReadInteger; // первый элемент пары
loop n - 1 do
begin
  var b := ReadInteger; // второй элемент пары
  if (a mod 7 = 0) and (b mod 17 <> 0)
      or (b mod 7 = 0) and (a mod 17 <> 0) then
        begin
          count += 1;
          minS := min(minS, a + b)
        end;
  a := b
end;
print(count, minS)